conformance/src/main/scala/za/co/absa/enceladus/conformance/CmdConfig.scala

```diff
@@ -30,21 +29,19 @@ import za.co.absa.enceladus.utils.fs.FileSystemVersionUtils
  *       Even if a field is mandatory it needs a default value.
  */
 case class CmdConfig(datasetName: String = "",
-    datasetVersion: Int = 1,
-    reportDate: String = "",
-    reportVersion: Option[Int] = None,
-    menasCredentials: Option[Either[MenasCredentials, CmdConfig.KeytabLocation]] = None,
-    performanceMetricsFile: Option[String] = None,
-    publishPathOverride: Option[String] = None,
-    folderPrefix: Option[String] = None,
-    experimentalMappingRule: Option[Boolean] = None,
-    isCatalystWorkaroundEnabled: Option[Boolean] = None,
-    autocleanStandardizedFolder: Option[Boolean] = None)
+                     datasetVersion: Int = 1,
+                     reportDate: String = "",
+                     reportVersion: Option[Int] = None,
+                     menasCredentials: Option[MenasCredentials] = None,
```

Comment:
As I understand `scopt` requires default values for each parameter. But we want to distinguish a case when no credentials are provided. So either:
* Implement something like `NullCredentials` as a default value (see Null Object design pattern) or
* Use `Option`

I like the options approach better since it does not introduce more names.
@GeorgiChochov, this is why I'd use an Option, not sure if your reasoning is the same.

Reply:
I left it an `Option` because it used to be an `Option`, but @yruslan is right about the reasoning of why it was initially an `Option`.
I've refactored it using a `Null Object`, seems like a cleaner solution to me.